packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart

```diff
@@ -141,19 +144,55 @@ class MethodChannelShare extends SharePlatform {
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
-  }) {
-    final mimeTypes =
-        files.map((e) => e.mimeType ?? _mimeTypeForPath(e.path)).toList();
+  }) async {
+    final filesWithPath = await _getFiles(files);
+
+    final mimeTypes = filesWithPath
+        .map((e) => e.mimeType ?? _mimeTypeForPath(e.path))
+        .toList();
 
     return shareFilesWithResult(
-      files.map((e) => e.path).toList(),
+      filesWithPath.map((e) => e.path).toList(),
       mimeTypes: mimeTypes,
       subject: subject,
       text: text,
       sharePositionOrigin: sharePositionOrigin,
     );
   }
 
+  /// if file doesn't contain path
+  /// then make new file in TemporaryDirectory and return with path
+  ///
+  /// the system will automatically delete files in this
+  /// TemporaryDirectory as disk space is needed elsewhere on the device
+  Future<List<XFile>> _getFiles(List<XFile> files) async {
+    int count = 1;
+
+    if (!files.every((element) => element.path.isNotEmpty)) {
+      final fileWithPath = files.where((element) => element.path.isNotEmpty);
+      final fileWithOutPath = files.where((element) => element.path.isEmpty);
+
+      final newFiles = <XFile>[];
+
+      final String tempPath = (await getTemporaryDirectory()).path;
+
+      for (final XFile element in fileWithOutPath) {
+        final path = '$tempPath/${element.name}$count';
+        final file = File(path);
+
+        await file.writeAsBytes(await element.readAsBytes());
+
+        newFiles.add(XFile(path));
+
+        count++;
+      }
+
+      return [...fileWithPath, ...newFiles];
```

Comment:
This would change the order of the files, and the provided mime types array wouldn't match again

Reply:
maintained files order